BaseLib/StringTools.cpp

```diff
@@ -141,3 +142,20 @@ void correctScientificNotation(std::string filename, std::size_t precision)
     rename(tmpFilename.c_str(), filename.c_str());
 }
 #endif
+
+
+std::string format(const char* format_str, ... )
+{
+    char buffer[256];
+    int size_of_buffer = sizeof(buffer) / sizeof(char);
+    memset(buffer, 0, sizeof(buffer));
+    va_list args;
+    va_start(args, format_str);
+#ifdef MSVC
+    vsnprintf_s(buffer, size_of_buffer, _TRUNCATE, format_str.data(), args);
+#else
+    vsnprintf(buffer, size_of_buffer, format_str, args );
```

Comment:
Error messages of more than 256 characters are pretty long, but I think there should be a check if the buffer was long enough, and if not a bigger buffer should be allocated dynamically (cf. "Notes" in http://en.cppreference.com/w/cpp/io/c/fprintf).